projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts

```diff
@@ -38,25 +37,29 @@ export class AnnotationSetComponent {
   }
 
   public onAnnotationUpdate(annotation: Annotation) {
-    const annotations = this.annotationSet.annotations.filter(a => a.id !== annotation.id);
-
-    annotations.push(annotation);
+    this.api
+      .postAnnotation(annotation)
+      .subscribe(newAnnotation => {
+        const index = this.annotationSet.annotations.findIndex(a => a.id === newAnnotation.id);
```

Comment:
these state updates feel like they should be done in the service as a result of the api call

Reply:
I'm not sure I agree, the API client is dumb it shouldn't really do anything apart from handle comms with the API.

Reply:
After discussion it was agreed the best course of action is to keep the API class as an ApiClient and create a Service that manages the state updates of an AnnotationSet. I'll create a refactor ticket for that.

Reply:
potential orphan subscription. might want an ngOnDestroy to tidy these up